### ISSUE TITLE
Dragging item from row X over row Y, then "out" (to remove) gives row Y as "container"

### ISSUE THREAD

**Issue by nickleus** [bevacqua/dragula#211]
v3.3.0 n previous v.

i found an interesting problem/bug where i have 2 rows/containers stacked vertically--row 0 above row 1.

when i click n drag an item from row 0 down over row 1, then continue dragging further down to a non-container area, in order to remove the item, "on.remove"'s "container" function param references row 1 instead of row 0 (the actual source container).

"el" is already disconnected from its source container, once inside "on.remove", so there's no way to get the original source container.

would b nice to expand on.remove w a "source" param--where the item was originally lifted.

ideas?

**Comment by beatfreaker** [comment]
can you represent it in a demo?

**Comment by beatfreaker** [comment]
@nickleus can you represent it in a demo?

**Comment by bevacqua** [comment]
I'm open to adding a `source` parameter in the third position for `cancel`, `remove`, and `shadow` events.

**Comment by nickleus** [comment]
@bevacqua that would b great :) i assume my description was good enough, that u don't need a codepen, or?

**Comment by bevacqua** [comment]
not really

**Comment by beatfreaker** [comment]
@bevacqua 
- ` drake.emit('shadow', item, dropTarget);` has to updated to ` drake.emit('shadow', item, dropTarget,_source);`
- `drake.emit(_copy ? 'cancel' : 'remove', item, parent);` to `drake.emit(_copy ? 'cancel' : 'remove', item, parent, _source);`
- `drake.emit('cancel', item, _source);` and what about this?

**Comment by nickleus** [comment]
@bevacqua @beatfreaker 

just tried this for "remove" n it seems to have worked :)

API:
```
drake.on('remove', function(el, container, source){...
```

dragula.js:280
```
drake.emit(_copy ? 'cancel' : 'remove', item, parent, _source);
```

**Comment by beatfreaker** [comment]
@nickleus are you working on a PR or i should submit one?

**Comment by nickleus** [comment]
@beatfreaker b my guest :)

**Comment by beatfreaker** [comment]
@bevacqua please check this [PR](https://github.com/bevacqua/dragula/pulls)

**Comment by bevacqua** [comment]
Implemented in `3.4.0`

**Comment by nickleus** [comment]
thx to @beatfreaker n @bevacqua :)
